index.js

```diff
@@ -163,39 +174,48 @@ function fastifyStatic (fastify, opts, next) {
       }
     } else {
       const globPattern = typeof opts.wildcard === 'string' ? opts.wildcard : '**/*'
-      glob(path.join(sendOptions.root, globPattern), { nodir: true }, function (err, files) {
-        if (err) {
-          return next(err)
-        }
-        const indexDirs = new Set()
-        const indexes = typeof opts.index === 'undefined' ? ['index.html'] : [].concat(opts.index || [])
-        for (let file of files) {
-          file = file.replace(sendOptions.root.replace(/\\/g, '/'), '').replace(/^\//, '')
-          const route = encodeURI(prefix + file).replace(/\/\//g, '/')
-          fastify.get(route, routeOpts, function (req, reply) {
-            pumpSendToReply(req, reply, '/' + file)
-          })
 
-          if (indexes.includes(path.posix.basename(route))) {
-            indexDirs.add(path.posix.dirname(route))
+      function globber (rootPath, cb) {
+        glob(path.join(rootPath, globPattern), { nodir: true }, function (err, files) {
+          if (err) {
+            return cb(err)
           }
-        }
-        indexDirs.forEach(function (dirname) {
-          const pathname = dirname + (dirname.endsWith('/') ? '' : '/')
-          const file = '/' + pathname.replace(prefix, '')
+          const indexDirs = new Set()
+          const indexes = typeof opts.index === 'undefined' ? ['index.html'] : [].concat(opts.index || [])
+          for (let file of files) {
+            file = file.replace(rootPath.replace(/\\/g, '/'), '').replace(/^\//, '')
+            const route = encodeURI(prefix + file).replace(/\/\//g, '/')
+            fastify.get(route, routeOpts, function (req, reply) {
+              pumpSendToReply(req, reply, '/' + file, rootPath)
+            })
 
-          fastify.get(pathname, routeOpts, function (req, reply) {
-            pumpSendToReply(req, reply, file)
-          })
+            if (indexes.includes(path.posix.basename(route))) {
+              indexDirs.add(path.posix.dirname(route))
+            }
+          }
+          indexDirs.forEach(function (dirname) {
+            const pathname = dirname + (dirname.endsWith('/') ? '' : '/')
+            const file = '/' + pathname.replace(prefix, '')
 
-          if (opts.redirect === true) {
-            fastify.get(pathname.replace(/\/$/, ''), routeOpts, function (req, reply) {
-              pumpSendToReply(req, reply, file.replace(/\/$/, ''))
+            fastify.get(pathname, routeOpts, function (req, reply) {
+              pumpSendToReply(req, reply, file, rootPath)
             })
-          }
+
+            if (opts.redirect === true) {
+              fastify.get(pathname.replace(/\/$/, ''), routeOpts, function (req, reply) {
+                pumpSendToReply(req, reply, file.replace(/\/$/, ''), rootPath)
+              })
+            }
+          })
+          cb()
         })
-        next()
-      })
+      }
+
+      if (Array.isArray(sendOptions.root)) {
+        async.eachSeries(sendOptions.root, globber, next)
```

Comment:
I would actually prefer to use `util.promisify` to get a promise based version of `glob` and use async/await in a for loop instead.

Reply:
No problem, I will update.